src/main/java/org/opensearch/ad/rest/RestValidateDetectorAction.java

```diff
@@ -1,44 +1,130 @@
+    private final NamedXContentRegistry xContentRegistry;
 
+    public static final Set<String> ALL_VALIDATION_ASPECTS_STRS = Arrays
+            .asList(ValidationAspect.values())
+            .stream()
+            .map(aspect -> aspect.getName())
+            .collect(Collectors.toSet());
+    public RestValidateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtension extension){
+        this.xContentRegistry = runner.getNamedXContentRegistry().getRegistry();
+        this.sdkClient = extension.getClient();
+    }
     @Override
     public List<Route> routes() {
         return List.of(new Route(POST, "/detectors/_validate"), new Route(POST, "/detectors/_validate/{type}"));
     }
 
     @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
+    public ExtensionRestResponse handleRequest(ExtensionRestRequest request){
         if (!EnabledSetting.isADPluginEnabled()) {
             throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
         }
-        Method method = request.method();
-
-        if (!Method.POST.equals(method)) {
-            return new ExtensionRestResponse(
-                request,
-                NOT_FOUND,
-                "Extension REST action improperly configured to handle " + request.toString()
-            );
+        AnomalyDetector detector;
+        XContentParser parser;
+        XContentBuilder builder = null;
+        ValidateAnomalyDetectorRequest validateAnomalyDetectorRequest;
+        try {
+            parser= request.contentParser(this.xContentRegistry);
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            String typesStr = validateTypeString(request);
+            DetectorValidationIssue issue=null;
+            try{
+                detector= AnomalyDetector.parse(parser);
+                //TODO Dependent on https://github.com/opensearch-project/opensearch-sdk-java/issues/211
+//                validateAnomalyDetectorRequest= new ValidateAnomalyDetectorRequest(
+//                        detector,
+//                        typesStr,
+//                        extension,
+//                        maxMultiEntityDetectors,
+//                        maxAnomalyFeatures,
+//                        requestTimeout
+//                );
+            }catch (Exception e){
+                if(e instanceof ADValidationException){
+                    ADValidationException ADException = (ADValidationException) e;
+                    issue = new DetectorValidationIssue(
+                            ADException.getAspect(),
+                            ADException.getType(),
+                            ADException.getMessage()
+                    );
+                }
+            }
+
+            try {
+                builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                builder.startObject();
+                builder.field("issue", issue);
```

Comment:
Why are we adding `DetectorValidationIssue` as a part of the builder response?

Reply:
We are adding DetectorValidationIssue in builder response as when there is a ADValidationexception it sends issue object in response which in turns what exactly was the issue in parsing. When there is no error then issue object remains null. In the plugin arch it is like when there is an error then return issue object and in case of successful execution no issue object is sent in response.

Reply:
The full response will be part of [SDK #224](https://github.com/opensearch-project/opensearch-sdk-java/issues/224).  The intent of issue 222 was just to create placeholders here.  This is a good start.